source/PlayerInfo.cpp

```diff
@@ -2925,6 +2959,22 @@ void PlayerInfo::RegisterDerivedConditions()
 		return rff.first - rff.second;
 	});
 
+	auto &&systemAttractionProvider = conditions.GetProviderPrefixed("attraction in system: ");
+	auto systemAttractionFun = [this](const string &name) -> double
+	{
+		const System *system = GameData::Systems().Find(name.substr(strlen("attraction in system: ")));
+		if(!system)
+			return 0;
+
+		double totalAttraction = 0;
+		for(const auto &raidFleet : system->GetGovernment()->RaidFleets())
+			totalAttraction += RaidFleetAttraction(raidFleet, system);
+		
+		return round(totalAttraction * 1000.);
+	};
+	systemAttractionProvider.SetGetFunction(systemAttractionFun);
+	systemAttractionProvider.SetHasFunction(systemAttractionFun);
```

Comment:
I've realized that this doesn't really work with multiple fleets. Consider the following scenario:
* System X has a single raid fleet while system Y has two raid fleets.
* In X, the fleet's attraction toward the player is 30. In Y, each fleet has an attraction of 15.
* The actual likelihood of any single fleet attacking you is `p(x) = 1 - (1 - x/1000)^10` (in English, one minus the probability for 10 of 10 attempts to spawn this fleet to fail) where x is the attraction value from this condition, as x is actually the odds of any one instance of the raid fleet attacking, and up to 10 raid fleets of a single type can be spawned at once.
* For system X, p(30) = 26.26%.
* For system Y, it's a little more complicated. We can't just take 2 * p(15), as that's not how these probabilistic events add together. What we actually have to do is take `1 - (1 - p(15)) * (1 - p(15))`, or `1 - (1 - p(15))^2` (in English, one minus the odds that the first fleet doesn't attack times the odds that the second fleet doesn't attack), which comes out to 26.09%. 
* So while both of these systems would return the same attraction value, they don't have the same chance of encountering a raid.

As such, assuming I haven't messed up my probabilities (please double check me), I think this needs to just return the raid chance as follows.
```suggestion
	auto &&systemAttractionProvider = conditions.GetProviderPrefixed("raid chance in system: ");
	auto systemAttractionFun = [this](const string &name) -> double
	{
		const System *system = GameData::Systems().Find(name.substr(strlen("raid chance in system: ")));
		if(!system)
			return 0.;

		// This variable represents the probability of no raid fleets spawning.
		double chance = 1.;
		for(const auto &raidFleet : system->GetGovernment()->RaidFleets())
		{
			// The attraction is the % chance for a single instance of this fleet to appear.
			double attraction = RaidFleetAttraction(raidFleet, system);
			// Calculate the % chance for no instances to appear from 10 rolls.
			double prob = pow(1. - attraction, 10.);
			// The chance of neither of two fleets appearing is the chance of the first not appearing
			// times the chance of the second not appearing.
			chance *= prob;
		}
		// The probability of any single fleet appearing is 1 - chance.
		return round((1. - chance) * 1000.);
	};
	systemAttractionProvider.SetGetFunction(systemAttractionFun);
	systemAttractionProvider.SetHasFunction(systemAttractionFun);
```

Reply:
Oh I didnt understand you meant to return the probability of attack itself, makes sense though

Reply:
Shouldnt pow check for (100-attraction) given we check attraction against a real that is between 0 and 100 (I think, right?)